libdns.go

```diff
@@ -105,26 +147,87 @@ type ZoneLister interface {
 }
 
 // Record is a generalized representation of a DNS record.
-//
-// The values of this struct should be free of zone-file-specific syntax,
-// except if this struct's fields do not sufficiently represent all the
-// fields of a certain record type; in that case, the remaining data for
-// which there are not specific fields should be stored in the Value as
-// it appears in the zone file.
```

Comment:
Rationale: It seems like most of the RR types _do_ expect a value encoded using zone-file-specific syntax, with only a few special cases that use other struct fields.

Reply:
Hm... yeah... the more I work with this the more I feel like Value should be the one and only "source of truth" for a record's value. I agree with this.